lib/suite-builder.js

```diff
@@ -0,0 +1,154 @@
+'use strict';
```

Comment:
I moved suite-builder to a separate file because it made a lot more sense to me for it to be extracted since its functionality is independent. I wanted `gemini.js` to be able to modify the SuiteBuilder, since that is the purpose of the `decorateSuite` function.

I'm also not sure how else I could structure this - where would I store the custom functions? If I store it on the gemini instance that plugins are passed, I'm not sure how to get a reference to that instance in the SuiteBuilder